jupyterlab_git/git.py

```diff
@@ -16,7 +18,14 @@
 # See https://git-scm.com/docs/git-config#_syntax for git var syntax
 CONFIG_PATTERN = re.compile(r"(?:^|\n)([\w\-\.]+)\=")
 DEFAULT_REMOTE_NAME = "origin"
+# Ensure on NFS or similar, that we give the .git/index.lock time to be removed
+MAX_WAIT_FOR_LOCK_S = 5
```

Comment:
Upon better review this one is not used.

Reply:
This should have been used. Fixed. The idea is you have 20s to wait for your turn to execute, and when you run, it will wait up to 5s for the `index.lock`.